magicbot/magicrobot.py

```diff
@@ -17,6 +17,49 @@
 
 __all__ = ['MagicRobot']
 
+def feedback(key=None):
+    """
+    If this decorator is applied to a function,
+    it's return value will automatically be sent
+    to NetworkTables at key ``/robot/component/component/key``
+
+    ``key`` is an optional parameter, and if it is not supplied,
+    the key will default to the method name with 'get_' removed.
+    If the method does not start with 'get_', the key will be the full
+    name of the method
+
+    The functions will automatically be called in disabled,
+    autonomous, and teleop.
+
+    The functions should only act as a getter, and accept
+    no arguments
+
+    Example
+
+    class Component1:
+
+        @feedback()
+        def get_angle(self):
+            return self.navx.getYaw()
+
+
+    In this example, the NetworkTable key is stored at
+    ``/robot/components/component1/angle``
+    """
+    def decorator(func):
+        nt_key = key
+        if nt_key is None:
+            # If no key is passed, get key from method name.
+            # -1 instead of 1 in case 'get_' is not present,
+            # in which case the key will be the method name
+            nt_key = func.__name__.split('get_')[-1]
+        # Set '__feedback__ attribute to be checked during injection
+        setattr(func, '__feedback__', True)
+        # Store key within the function to avoid using class dictionary
+        setattr(func, '__key__', nt_key)
+        return func
```

Comment:
It doesn't *need* to be but there typically aren't reasons for it not to be, so we should enforce it.